### DISCUSSION TITLE
What makes this Queue still useful after @isolated(any) in Swift 6,

### DISCUSSION THREAD

**Discussion by sumuhans** [dimasusername/Queue#6] [Q&A]
If my understanding is correct, Swift 6 now ensures Tasks will be executed in the order they are created, as long as the Tasks are all created in the same "context" (In the same actor or in outside of any actor (in which case they will be executed in the global executer. I guess?) So in any case, if we have a bunch of Tasks like this
```
Task { print("a") }
Task { print("b") }
Task { print("c") }
```
Then they will always be executed in the right order. As mentioned [here](https://www.massicotte.org/ordering-and-concurrency) 
The article says 
"...Update: Swift 6 has strengthened the ordering guarantees of Task via the [@isolated(any)](https://www.massicotte.org/concurrency-swift-6-se-0431) attribute. This makes a big difference, but the general problems here are still applicable..." 
Please can someone explain what those  "general problems" are that are  "still applicable"?
I guess that order will only be FIFO if the priorities are equal. So is any any advantage in using this Queue unless we need to execute blocks with different priorities and still want to preserve order?

**Comment by mattmassicotte** [comment]
Good question! It's true that **if** you can create the `Task` instances synchronously, they will preserve ordering. There are a handful of situations where I have used a queue structure that I would not longer need to with Swift 6.

But if you cannot create the tasks synchronously, the changes to the `Task` API will not help.

If you need to preserve FIFO ordering, you have to think about how it's going to happen. It could be that the changes to Swift 6 will do it. But, if not, you'll still need a queue of some kind. However, often a plan old `AsyncStream` can work. You may not need the features of this particular implementation.

**Reply by sumuhans** [reply]
Thank you so much @mattmassicotte for taking the time to respond. It is very kind of you. Just to make sure I understand what you meant, please could you provide small snippets to illustrate each of the two cases you mentioned below please?  
>  It's true that **if** you can create the `Task` instances synchronously, they will preserve ordering. There are a handful of situations where I have used a queue structure that I would not longer need to with Swift 6.
> 
> But if you cannot create the tasks synchronously, the changes to the `Task` API will not help.
>

**Reply by mattmassicotte** [reply]
I thought about this for a little bit, and it's actually kinda hard to do! If the new Task semantics work enough for you, that's great. They are definitely better. I should really spend more time coming up with better examples of when queues are useful, though.
